modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -307,85 +322,114 @@ package object widgets {
     }
   }
 
-  abstract class Chart[C:ToPoints:Sampler] extends JsWorld[Seq[(String, Any)], Seq[(String, Any)]] {
-    import notebook.JSBus._
-
-    def originalData:C
-    def maxPoints:Int
-
+  abstract class OriginalDataToSeqConversions[C: ToPoints : Sampler](originalData: C, maxPoints: Int)
```

Comment:
@huitseeker see this refactoring too  https://github.com/andypetrella/spark-notebook/pull/624/commits/6af7788c90f4dd6a4312a0ed112af384af101a74
IMHO the type-alias doesn't make it worse ;)